### ISSUE TITLE
AWS deployment requires terraform >= 0.12

### ISSUE THREAD

**Issue by shinnosuke-okada** [pingcap/tidb-operator#635]
## Bug Report

**What version of Kubernetes are you using?**
N/A

**What version of TiDB Operator are you using?**
latest master

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**
N/A

**What's the status of the TiDB cluster pods?**
N/A

**What did you do?**
1. Checkout latest tidb-operator
2. `terraform apply` from `deploy/aws`

**What did you expect to see?**
Successful deployment according to the README
terraform version requirement should be documented

**What did you see instead?**
Parsing error from terraform due to first-class expression syntax. For example, in `deploy/aws/bastion.tf`, following lines will not work with terraform versions v0.11 and lower:
```
  vpc_id      = var.create_vpc ? module.vpc.vpc_id : var.vpc_id
```
```
  instance_count              = var.create_bastion ? 1 : 0
```
Looks like this feature was introduced in [terraform v0.12](https://www.hashicorp.com/blog/announcing-terraform-0-12).

**Comment by shinnosuke-okada** [comment]
@aylei FYI - seems like a new requirement from https://github.com/pingcap/tidb-operator/pull/616

**Comment by gregwebs** [comment]
closed by #636
